ExampleMod/Content/Projectiles/ExampleSineWaveMotionProjectile.cs

```diff
@@ -0,0 +1,150 @@
+		}
+
+		// This projectile updates its position manually
+		public override bool ShouldUpdatePosition() {
+			return false;
+		}
+
+		public override void OnSpawn(IEntitySource source) {
+			initialCenter = Projectile.Center;
+		}
+
+		public override Color? GetAlpha(Color lightColor) {
+			return drawColor * 0.64f;
+		}
+
+		public override void AI() {
+			// Ensure that the "waveDirection = 1" projectile starts above the "waveDirection = -1" projectile not matter which direction they were fired from
+			int trueDirection = Projectile.direction * waveDirection;
+
+			// How many oscillations happen per second
+			// Higher value = more oscillations
+			float wavesPerSecond = 2;
+
+			// Math.Sin expects a radians angle instead of degrees
+			float radians = MathHelper.ToRadians(sineTimer * 6f * wavesPerSecond);
+			float sine = (float)Math.Sin(radians) * trueDirection;
+
+			// Using the calculated sine value, generate an offset used to position the projectile on the wave
+			// The offset should be perpendicular to the velocity direction, hence the RotatedBy call
+			Vector2 offset = Projectile.velocity.SafeNormalize(Vector2.UnitX).RotatedBy(MathHelper.PiOver2 * -1);
+
+			// How wide the wave should be, times two
+			// An amplitude of 24 pixels is 1.5 tiles, meaning the total wave width is 48 pixels, or 3 tiles
+			float waveAmplitude = 24;
+
+			// Get the offset used to adjust the projectile's position
+			offset *= sine * waveAmplitude;
+
+			// Update the position manually since ShouldUpdatePosition returns false
+			initialCenter += Projectile.velocity;
+			Projectile.Center = initialCenter + offset;
+
+			// Update the rotation used to draw the projectile
+			// This projectile should act as if it were moving along the sine wave
+			float cosine = (float)Math.Cos(radians) * trueDirection;
+			Projectile.rotation = Projectile.velocity.ToRotation() + MathHelper.PiOver4 * cosine * -1;
+
+			// Update the frame used to draw the projectile
+			const float cosineOf45Degrees = 0.707106781f;
+			float cosineAbsolute = Math.Abs(cosine);
+			if (sine > 0) {
+				// Offset is counter-clockwise to the direction
+				if (cosineAbsolute > cosineOf45Degrees) {
+					Projectile.frame = Projectile.direction == 1 ? 0 : 2;
+				}
+				else {
+					Projectile.frame = 1;
+				}
+			}
+			else if (sine < 0) {
+				// Offset is clockwise to the direction
+				if (cosineAbsolute > cosineOf45Degrees) {
+					Projectile.frame = Projectile.direction == 1 ? 2 : 0;
+				}
+				else {
+					Projectile.frame = 1;
+				}
+			}
+			else {
+				// Offset is zero
+				Projectile.frame = 1;
+			}
+
+			// Spawn dusts
+			Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.ExampleSolution>(), Velocity: Vector2.Zero, newColor: drawColor, Scale: 0.5f);
+
+			Projectile.direction = (Projectile.velocity.X > 0).ToDirectionInt();
+
+			sineTimer++;
+		}
+
+		public override bool PreDraw(ref Color lightColor) {
```

Comment:
I'm pretty sure this shouldn't be required. Why is positioning and framing the texture not enough? By altering the draw here, you can create differences between draw and hitbox.

Reply:
i tried to do that via the DrawOffset variables, but it was annoying to make them work when the projectile was reversed